include/nanogui/window.h

```diff
@@ -62,11 +69,18 @@ class NANOGUI_EXPORT Window : public Widget {
 protected:
     /// Internal helper function to maintain nested window position values; overridden in \ref Popup
     virtual void refreshRelativePlacement();
+
+    virtual int checkHorizontalResize(const Vector2i &mousePos);
+    virtual int checkVerticalResize(const Vector2i &mousePos);
 protected:
     std::string mTitle;
     Widget *mButtonPanel;
     bool mModal;
     bool mDrag;
+    bool mResize;
```

Comment:
I think that the ``mResize`` variable is not needed. Instead of redoing the layout on every draw call (probably many more necessary), call ``performLayout`` when a motion event changes the size of the window. For this, you will need the NanoVG context, which can be acquired from the ``static_cast<Screen *>(parent())->nvgContext()``.